DesktopEdge/MainWindow.xaml.cs

```diff
@@ -646,16 +646,20 @@ public partial class MainWindow : Window {
 		}
 
 		private void ServiceClient_OnBulkServiceEvent(object sender, BulkServiceEvent e) {
-			var found = identities.Find(id => id.Fingerprint == e.Fingerprint);
+			var found = identities.Find(id => id.Identifier == e.Identifier);
 			if (found == null) {
-				logger.Warn($"{e.Action} service event for {e.Fingerprint} but the provided identity fingerprint was not found!");
+				logger.Warn($"{e.Action} service event for {e.Identifier} but the provided identity identifier was not found!");
 				return;
 			} else {
-				foreach (var removed in e.RemovedServices) {
-					removeService(found, removed);
+				if (e.RemovedServices != null) {

```

Comment:
did you actually get a null to hit this condition? this seems like a good thing to guard against - but having any 'null' removed services is a whole other bug that seems like should get tracked down